lib/jmap/core/reference_prefix.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:equatable/equatable.dart';
+import 'package:quiver/check.dart';
+
+class ReferencePrefix with EquatableMixin {
+
+  static final defaultPrefix = ReferencePrefix('#');
+
+  final String value;
+
+  ReferencePrefix(this.value) {
+    checkArgument(value.isNotEmpty, message: 'invalid length');
+    checkArgument(value.length < 255, message: 'invalid length');
```

Comment:
prefix can be other different `#`?

Reply:
Of course

Reply:
IMO, we have only `#` to indicate it is reference. So no need to create the class for it. It will not be change.

Reply:
IMO We should create such a class so that we can extend the system more later. Can be used for other systems.